[#cOeut iqt/1028 fix discreetly v0.1.2]

minhducvt293 opened the pull request:
* **Trello Ticket**: https://trello.com/c/cOeutIqt/1028-fix-discreetly
* Other Relevant Links [google-cloud-kms 2.0.0 migration guide](https://googleapis.dev/python/cloudkms/latest/UPGRADING.html)
[discreetly v0.2.0 PR](https://github.com/tra-sg/discreetly/pull/10)

# What does this Pull Request do?
- pin `google-cloud-kms` version < 2.0.0
- add template for making commit and PRs

prepare to release `discreetly` v0.1.2

# How should this be validated?
1. uninstall `google-cloud-kms`
```console
pip uninstall google-cloud-kms
```

2. install dependencies:
```console
pip install -r requirements.txt
```

3. run pytest should be sufficient
```console
pytest
```

4. Follow the `README.md` and verify those functions work correctly.

# Checklist
<!-- This is a reminder checklist that you've taken all necessary steps to prepare this PR
Replace `[ ]` with `[x]` for all that apply.-->
- [x] You've fetched and rebased your branch on an **up-to-date** version `master`
- [x] PR contains new tests relevant for the feature/problem this PR is addressing
- [x] All tests and linting pass
- [x] Your commits have been squashed/rebased/etc. for reviewer clarity


# Additional Notes:
This PR is for release of v0.1.2 which supports `python` 2 and 3
Other PR (https://github.com/tra-sg/discreetly/pull/10) is for release of v0.2.0 which supports `python` 3 only

# Interested parties
@ghosalya as main reviewer
@eddies

ghosalya commented on .github/hooks/prepare-commit-msg:
```diff
@@ -0,0 +1,69 @@
+# letters of <shortLink>.
+#
+# The second form allows you to copy-and-paste the last part of a Trello card url following
+# "https://trello.com/c/" to use as the branch name, e.g.
```
we don't want trello stuff to be part of the template right?

ghosalya commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,48 @@
+* **Trello Ticket**: <!-- insert URL -->
```
Same comment on trello

ghosalya commented on .github/workflows/tests.yml:
```diff
@@ -6,21 +6,25 @@ jobs:
         with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/setup.py') }}
+          path: ${{ steps.pip-cache.outputs.dir }}/../../../
```
did we figure out why this ends up working? It will be useful to have a comment on why this is a correct path (or a link to some discussions)

ghosalya commented:
@minhducvt293 also, can you add `.direnv/` and `.vscode` to the `.gitignore` file?

ghosalya reviewed (changes requested):
lgtm, just need to squash

ghosalya reviewed (changes requested):
we need an update in CHANGELOG.md if this is to be released

ghosalya commented on CHANGELOG.md:
```diff
@@ -9,6 +9,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+
+### Added
+
+- Pin down `google-cloud-kms` to `1.4.0`
```
since this is a changelog since `v0.1.1`, it would be appropriate to also include changes from https://github.com/tra-sg/discreetly/pull/8

The current changelog seems to refer to issues, but some of these changes aren't from an existing issue. I suggest this entry should go something like this (using links to PR instead of issues)

```
- Pin down `google-cloud-kms` to `1.4.0`, and update `pip cache` for tests ([#11])
- chore: Fix setup.py description and README.md release instructions ([#8])
```

minhducvt293 committed:
[#cOeutIqt] 1028 fix discreetly v0.1.2
- Pin down `google-cloud-kms` to `1.4.0`
- Update `.envrc` and `README.md`
- Update `pip cache` for `pypi` tests
- Add template for PRs and commits

ghosalya reviewed (approved):
lgtm
